### PR TITLE
Add option to sentAt field to be null

### PR DESCRIPTION
**Pull request by edsonjab**
Override method create on Batch class to lend to user define his own date on sentAt field or use a null value.

Issue #340

### CODE REVIEW

**Comment by pooyaj** [comment]
@edsonjab please check out the comment on the issue: https://github.com/segmentio/analytics-java/issues/340

**Review by wenxi-zeng — CHANGES_REQUESTED** [review]
the changes does not address the issue, since `Batch.create()` is only used internally to create a batch. 

you'll need to do the following:
1. in `Message` add a new nullable field `sentAt`
2. in `MessageBuilder` add a new field `sentAt` and make changes accordingly. you can take `userId` as an example for what needs to be done
3. in `MessageBuilder.realBuild` add a new parameter `sentAt` and make changes accordingly. 
4. discard all your changes in `Batch`. in `create` method do the following
```java
if any message in batch has `sentAt` not null
    use that value to create batch
otherwise
    use `new Date()` to create batch
```

now the users can use `builder.sentAt()` to modify the value

File: analytics-core/.factorypath [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<factorypath>
```
Comment (wenxi-zeng) [inline analytics-core/.factorypath]: is this one accidentally checked in?

File: analytics-core/src/main/java/com/segment/analytics/messages/Batch.java [hunk 2]
```diff
@@ -6,18 +6,27 @@
...
+    return new AutoValue_Batch(
+        batch,
+        sentAtNull == null ? new Date() : sentAtNull.sentAt(),
+        context,
+        SEQUENCE_GENERATOR.incrementAndGet());
   }
 
   public abstract List<Message> batch();
 
+  @Nullable
```
Comment (wenxi-zeng) [inline analytics-core/src/main/java/com/segment/analytics/messages/Batch.java:29]: no need to be nullable here

File: analytics-core/src/test/java/com/segment/analytics/messages/BatchTest.java [hunk 3]
```diff
@@ -26,6 +28,22 @@ public void create(TestUtils.MessageBuilderFactory factory) {
     assertThat(batch.context()).isEqualTo(context);
   }
 
+  @Test
+  public void createWithSentAt(TestUtils.MessageBuilderFactory factory) throws ParseException {
```
Comment (wenxi-zeng) [inline analytics-core/src/test/java/com/segment/analytics/messages/BatchTest.java:32]: better to have another test with messages has `sentAt` as nulls, so we have both paths covered.

**Review by wenxi-zeng — APPROVED** [review]
rest LGTM

**Review by pooyaj — APPROVED** [review]

**Pull request merged**